app/derivative_services/image_derivative_service.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+class ImageDerivativeService
+  class Factory
+    attr_reader :adapter, :repository
+    def initialize(adapter, repository)
+      @adapter = adapter
+      @repository = repository
+    end
+
+    def new(file_set)
+      ::ImageDerivativeService.new(file_set, original_file(file_set), adapter, repository)
+    end
+
+    def original_file(file_set)
+      members(file_set).find { |x| x.use.include?("original") }
+    end
+
+    def members(file_set)
+      adapter.query_service.find_members(model: file_set)
+    end
+  end
+  attr_reader :file_set, :original_file, :adapter, :repository
+  delegate :mime_type, to: :original_file
+  delegate :persister, to: :adapter
+  def initialize(file_set, original_file, adapter, repository)
+    @file_set = file_set
+    @original_file = original_file
+    @adapter = adapter
+    @repository = repository
+  end
+
+  def create_derivatives
+    Hydra::Derivatives::ImageDerivatives.create(filename,
+                                                outputs: [{ label: :thumbnail, format: 'jpg', size: '200x150>', url: URI("file://#{temporary_output.path}") }])
```

Comment:
We've overridden a lot of aspects of derivative creation in LAKE, @escowles. I'm not sure if this is meeting your point - putting thumbnail sizes in a method might be a very beneficial thing - but here's a working example. Might not be the best way, but food for thought.

https://github.com/aic-collections/aicdams-lakeshore/blob/master/app/models/file_set.rb